tests/terraform/createcluster.go

```diff
@@ -0,0 +1,61 @@
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"path/filepath"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+)
+
+var destroy = flag.Bool("destroy", false, "a bool")
+var nodeOs = flag.String("node_os", "centos8", "a string")
+var externalDb = flag.String("external_db", "mysql", "a string")
+var arch = flag.String("arch", "amd64", "a string")
+var clusterType = flag.String("cluster_type", "etcd", "a string")
+var resourceName = flag.String("resource_name", "etcd", "a string")
+var sshuser = flag.String("sshuser", "ubuntu", "a string")
+var sshkey = flag.String("sshkey", "", "a string")
+
+var (
+	kubeConfigFile string
+	masterIPs      string
+	workerIPs      string
+)
+
+func BuildCluster(nodeOs, clusterType, externalDb, resourceName string, t *testing.T, destroy bool) (string, string, string) {
+
+	tDir := "./modules/k3scluster"
+	vDir := "/config/" + nodeOs + clusterType + ".tfvars"
+
+	if externalDb != "" {
+		vDir = "/config/" + nodeOs + externalDb + ".tfvars"
+	}
+
+	tfDir, _ := filepath.Abs(tDir)
+	varDir, _ := filepath.Abs(vDir)
+	TerraformOptions := &terraform.Options{
+		TerraformDir: tfDir,
+		VarFiles:     []string{varDir},
+		Vars: map[string]interface{}{
+			"cluster_type":  clusterType,
+			"resource_name": resourceName,
+			"external_db":   externalDb,
+		},
+	}
+
+	if destroy {
+		fmt.Printf("Cluster is being deleted")
+		terraform.Destroy(t, TerraformOptions)
+		return "", "", ""
+	}
+
+	fmt.Printf("Creating Cluster")
+	terraform.InitAndApply(t, TerraformOptions)
+	kubeconfig := terraform.Output(t, TerraformOptions, "kubeconfig") + "_kubeconfig"
+	masterIps := terraform.Output(t, TerraformOptions, "master_ips")
```

Comment:
Nit:

```suggestion
	masterIPs := terraform.Output(t, TerraformOptions, "master_ips")
```

Same would go for the variable name below.

Reply:
Updated